pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -50,59 +55,31 @@ const (
 
 // GetVSphere reads vSphere configuration from system environment and construct vSphere object
 func GetVSphere() (*VSphere, error) {
-	cfg := getVSphereConfig()
-	vSphereConn := getVSphereConn(cfg)
-	client, err := GetgovmomiClient(vSphereConn)
+	cfg, err := getVSphereConfig()
+	if err != nil {
+		return nil, err
+	}
+	vs, err := newControllerNode(*cfg)
 	if err != nil {
 		return nil, err
 	}
-	vSphereConn.GoVmomiClient = client
-	vsphereIns := &VSphereInstance{
-		conn: vSphereConn,
-		cfg: &VirtualCenterConfig{
-			User:              cfg.Global.User,
-			Password:          cfg.Global.Password,
-			VCenterPort:       cfg.Global.VCenterPort,
-			Datacenters:       cfg.Global.Datacenters,
-			RoundTripperCount: cfg.Global.RoundTripperCount,
-		},
-	}
-	vsphereInsMap := make(map[string]*VSphereInstance)
-	vsphereInsMap[cfg.Global.VCenterIP] = vsphereIns
-	// TODO: Initialize nodeManager and set it in VSphere.
-	vs := &VSphere{
-		vsphereInstanceMap: vsphereInsMap,
-		hostName:           "",
-		cfg:                cfg,
-		nodeManager: &NodeManager{
-			vsphereInstanceMap: vsphereInsMap,
-			nodeInfoMap:        make(map[string]*NodeInfo),
-			registeredNodes:    make(map[string]*v1.Node),
-		},
-	}
-	runtime.SetFinalizer(vs, logout)
 	return vs, nil
 }
 
-func getVSphereConfig() *VSphereConfig {
-	var cfg VSphereConfig
-	cfg.Global.VCenterIP = os.Getenv("VSPHERE_VCENTER")
-	cfg.Global.VCenterPort = os.Getenv("VSPHERE_VCENTER_PORT")
-	cfg.Global.User = os.Getenv("VSPHERE_USER")
-	cfg.Global.Password = os.Getenv("VSPHERE_PASSWORD")
-	cfg.Global.Datacenters = os.Getenv("VSPHERE_DATACENTER")
-	cfg.Global.DefaultDatastore = os.Getenv("VSPHERE_DATASTORE")
-	cfg.Global.WorkingDir = os.Getenv("VSPHERE_WORKING_DIR")
-	cfg.Global.VMName = os.Getenv("VSPHERE_VM_NAME")
-	cfg.Global.InsecureFlag = false
-	if strings.ToLower(os.Getenv("VSPHERE_INSECURE")) == "true" {
-		cfg.Global.InsecureFlag = true
-	}
-	cfg.Workspace.VCenterIP = cfg.Global.VCenterIP
-	cfg.Workspace.Datacenter = cfg.Global.Datacenters
-	cfg.Workspace.DefaultDatastore = cfg.Global.DefaultDatastore
-	cfg.Workspace.Folder = cfg.Global.WorkingDir
-	return &cfg
+func getVSphereConfig() (*VSphereConfig, error) {
+	if ConfFileLocation == "" {
```

Comment:
why not pass in confFileLocation or look up from env within this method?

Reply:
I have modified the look up for env in this method only. Please check the latest commit.

Reply:
thanks